### PR TITLE
MINOR: Make SenderMetrics class private with private members

### PR DESCRIPTION
**Pull request by rajatvig**
to reduce accessibility of the inner class.

### CODE REVIEW

**Comment by rajatvig** [comment]
The test failures have seemingly little to do with the changes.

**Comment by guozhangwang** [comment]
test this

**Comment by guozhangwang** [comment]
test this please

**Comment by guozhangwang** [comment]
test this please

File: clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java [hunk 1]
```diff
@@ -374,20 +374,20 @@ public void wakeup() {
     /**
      * A collection of sensors for the sender
      */
-    private class SenderMetrics {
+    private static class SenderMetrics {
```
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java:377]: Why we want to make it static? I feel it is not worthwhile as we now have to send client and metadata parameters into the class.

File: clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java [hunk 2]
```diff
@@ -488,8 +488,7 @@ private void maybeRegisterTopicMetrics(String topic) {
             }
         }
 
-        public void updateProduceRequestMetrics(Map<Integer, List<RecordBatch>> batches) {
-            long now = time.milliseconds();
+        public void updateProduceRequestMetrics(Map<Integer, List<RecordBatch>> batches, long now) {
```
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java:491]: We can also remove "public" here to make it package-private.

File: clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java [hunk 3]
```diff
@@ -523,26 +522,23 @@ public void updateProduceRequestMetrics(Map<Integer, List<RecordBatch>> batches)
...
             this.errorSensor.record(count, now);
             String topicErrorName = "topic." + topic + ".record-errors";
             Sensor topicErrorSensor = this.metrics.getSensor(topicErrorName);
             if (topicErrorSensor != null)
                 topicErrorSensor.record(count, now);
         }
 
-        public void recordLatency(String node, long latency) {
-            long now = time.milliseconds();
+        public void recordLatency(String node, long latency, long now) {
```
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java:541]: Ditto here.

**Review by guozhangwang — COMMENTED** [review]
Left a few comments.

**Comment by ijuma** [comment]
It would be good to explain what we're trying to achieve here.

**Pull request closed without merging**